test/blackbox/common/DDSBlackboxTestsSecurity.cpp

```diff
@@ -74,43 +103,77 @@ TEST(DDSSecurity, big_message_corner_case)
-    builder->set_name(TEST_TOPIC_NAME);
+            DynamicTypeBuilderFactory::get_instance()->create_array_builder(base_type, lengths);
+    builder->set_name(name);
     DynamicType_ptr array_type = builder->build();
 
     fastdds::TypeSupport type_support(new eprosima::fastrtps::types::DynamicPubSubType(array_type));
     type_support.get()->auto_fill_type_information(false);
     type_support.get()->auto_fill_type_object(false);
     EXPECT_EQ(ReturnCode_t::RETCODE_OK, participant->register_type(type_support));
 
-    auto topic = participant->create_topic(TEST_TOPIC_NAME, TEST_TOPIC_NAME, fastdds::TOPIC_QOS_DEFAULT);
+    auto topic = participant->create_topic(name, name, fastdds::TOPIC_QOS_DEFAULT);
     ASSERT_NE(nullptr, topic);
     auto subscriber = participant->create_subscriber(fastdds::SUBSCRIBER_QOS_DEFAULT);
     ASSERT_NE(nullptr, subscriber);
     auto reader = subscriber->create_datareader(topic, fastdds::DATAREADER_QOS_DEFAULT);
     ASSERT_NE(nullptr, reader);
     auto publisher = participant->create_publisher(fastdds::PUBLISHER_QOS_DEFAULT);
     ASSERT_NE(nullptr, publisher);
-    auto writer = publisher->create_datawriter(topic, fastdds::DATAWRITER_QOS_DEFAULT);
+
+    std::atomic_bool writer_matched{ false };
+    WriterListener wlistener(writer_matched);
+    auto writer = publisher->create_datawriter(topic, fastdds::DATAWRITER_QOS_DEFAULT, &wlistener,
+                    fastdds::StatusMask::publication_matched());
     ASSERT_NE(nullptr, writer);
 
-    // TODO: wait for discovery
-    std::this_thread::sleep_for(std::chrono::milliseconds(500));
+    // Wait for discovery to occur
+    while (!writer_matched)
+    {
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+    }
 
     auto data = eprosima::fastrtps::types::DynamicDataFactory::get_instance()->create_data(array_type);
     EXPECT_EQ(ReturnCode_t::RETCODE_OK, writer->write(data, fastdds::HANDLE_NIL));
 
     fastdds::SampleInfo info{};
-    do
+    bool taken = false;
+    for (size_t n_tries = 0; n_tries < 6u; ++n_tries)
     {
+        if (ReturnCode_t::RETCODE_OK == reader->take_next_sample(data, &info))
+        {
+            taken = true;
+            break;
+        }
         std::this_thread::sleep_for(std::chrono::milliseconds(500));
-    } while (ReturnCode_t::RETCODE_OK != reader->take_next_sample(data, &info));
+    }
 
+    EXPECT_TRUE(taken);
     EXPECT_EQ(ReturnCode_t::RETCODE_OK, participant->delete_contained_entities());
     fastdds::DomainParticipantFactory::get_instance()->delete_participant(participant);
 }
 
+TEST(DDSSecurity, big_message_corner_case)
+{
+    const uint32_t array_lengths[] =
+    {
+        // Working case
+        65200,
+        // Failing cases
+        65240, 65252, 65260, 65300,
+        // Working case
+        65400
+    };
+
+    std::string topic_name = TEST_TOPIC_NAME;
+    for (uint32_t len : array_lengths)
+    {
+        topic_name += "_";
```

Comment:
I suppose some more specific information related with this loop was intended here and it was forgotten at the last minute.

Reply:
I just wanted topic names to be different to avoid collision between tests, so just added a character. Taking a closer look now though, I think it may not be necessary as (1) the participant is created and destroyed on each iteration, and (2) it is whitelisting localhost, so collision will be avoided.